core2s.test_wrap_malloc_mimalloc failure

```
C:\emsdk\emscripten\main>test\runner core2s.test_wrap_malloc_mimalloc
Running 1 tests
[1/1] test_wrap_malloc_mimalloc (test_core.core2s.test_wrap_malloc_mimalloc) ... FAIL

Stdout:
-- begin program output --
Aborted(segmentation fault)
C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1
var Module=typeof Module!="undefined"?Module:{};var ENVIRONMENT_IS_WEB=!!globalThis.window;var ENVIRONMENT_IS_WORKER=!!globalThis.WorkerGlobalScope;var ENVIRONMENT_IS_NODE=globalThis.process?.versions?.node&&globalThis.process?.type!="renderer";var arguments_=[];var thisProgram="./this.program";var quit_=(status,toThrow)=>{throw toThrow};var _scriptName=globalThis.document?.currentScript?.src;if(typeof __filename!="undefined"){_scriptName=__filename}else if(ENVIRONMENT_IS_WORKER){_scriptName=self.location.href}var scriptDirectory="";function locateFile(path){if(Module["locateFile"]){return Module["locateFile"](path,scriptDirectory)}return scriptDirectory+path}var readAsync,readBinary;if(ENVIRONMENT_IS_NODE){var fs=require("node:fs");scriptDirectory=__dirname+"/";readBinary=filename=>{filename=isFileURI(filename)?new URL(filename):filename;var ret=fs.readFileSync(filename);return ret};readAsync=async(filename,binary=true)=>{filename=isFileURI(filename)?new URL(filename):filename;var ret=fs.readFileSync(filename,binary?undefined:"utf8");return ret};if(process.argv.length>1){thisProgram=process.argv[1].replace(/\\/g,"/")}arguments_=process.argv.slice(2);if(typeof module!="undefined"){module["exports"]=Module}quit_=(status,toThrow)=>{process.exitCode=status;throw toThrow}}else if(ENVIRONMENT_IS_WEB||ENVIRONMENT_IS_WORKER){try{scriptDirectory=new URL(".",_scriptName).href}catch{}{if(ENVIRONMENT_IS_WORKER){readBinary=url=>{var xhr=new XMLHttpRequest;xhr.open("GET",url,false);xhr.responseType="arraybuffer";xhr.send(null);return new Uint8Array(xhr.response)}}readAsync=async url=>{if(isFileURI(url)){return new Promise((resolve,reject)=>{var xhr=new XMLHttpRequest;xhr.open("GET",url,true);xhr.responseType="arraybuffer";xhr.onload=()=>{if(xhr.status==200||xhr.status==0&&xhr.response){resolve(xhr.response);return}reject(xhr.status)};xhr.onerror=reject;xhr.send(null)})}var response=await fetch(url,{credentials:"same-origin"});if(response.ok){return response.arrayBuffer()}throw new Error(response.status+" : "+response.url)}}}else{}var out=console.log.bind(console);var err=console.error.bind(console);var wasmBinary;var ABORT=false;var EXITSTATUS;var isFileURI=filename=>filename.startsWith("file://");class EmscriptenEH{}class EmscriptenSjLj extends EmscriptenEH{}function SAFE_HEAP_INDEX(arr,idx,action){const bytes=arr.BYTES_PER_ELEMENT;const dest=idx*bytes;if(idx<=0)abort(`segmentation fault ${action} ${bytes} bytes at address ${dest}`);if(runtimeInitialized){var brk=_sbrk(0);if(dest+bytes>brk)abort(`segmentation fault, exceeded the top of the available dynamic heap when ${action} ${bytes} bytes at address ${dest}. DYNAMICTOP=${brk}`);if(brk<_emscripten_stack_get_base())abort(`brk >= _emscripten_stack_get_base() (brk=${brk}, _emscripten_stack_get_base()=${_emscripten_stack_get_base()})`);if(brk>wasmMemory.buffer.byteLength)abort(`brk <= wasmMemory.buffer.byteLength (brk=${brk}, wasmMemory.buffer.byteLength=${wasmMemory.buffer.byteLength})`)}return idx}function segfault(){abort("segmentation fault")}function alignfault(){abort("alignment fault")}var runtimeInitialized=false;function updateMemoryViews(){var b=wasmMemory.buffer;HEAP8=new Int8Array(b);HEAP16=new Int16Array(b);HEAPU8=new Uint8Array(b);HEAPU16=new Uint16Array(b);HEAP32=new Int32Array(b);HEAPU32=new Uint32Array(b);HEAPF32=new Float32Array(b);HEAPF64=new Float64Array(b);HEAP64=new BigInt64Array(b);HEAPU64=new BigUint64Array(b)}function preRun(){if(Module["preRun"]){if(typeof Module["preRun"]=="function")Module["preRun"]=[Module["preRun"]];while(Module["preRun"].length){addOnPreRun(Module["preRun"].shift())}}callRuntimeCallbacks(onPreRuns)}function initRuntime(){runtimeInitialized=true;wasmExports["__wasm_call_ctors"]()}function preMain(){}function postRun(){if(Module["postRun"]){if(typeof Module["postRun"]=="function")Module["postRun"]=[Module["postRun"]];while(Module["postRun"].length){addOnPostRun(Module["postRun"].shift())}}callRuntimeCallbacks(onPostRuns)}function abort(what){Module["onAbort"]?.(what);what=`Aborted(${what})`;err(what);ABORT=true;what+=". Build with -sASSERTIONS for more info.";var e=new WebAssembly.RuntimeError(what);throw e}var wasmBinaryFile;function findWasmBinary(){return locateFile("test_wrap_malloc.wasm")}function getBinarySync(file){if(file==wasmBinaryFile&&wasmBinary){return new Uint8Array(wasmBinary)}if(readBinary){return readBinary(file)}throw"both async and sync fetching of the wasm failed"}async function getWasmBinary(binaryFile){if(!wasmBinary){try{var response=await readAsync(binaryFile);return new Uint8Array(response)}catch{}}return getBinarySync(binaryFile)}async function instantiateArrayBuffer(binaryFile,imports){try{var binary=await getWasmBinary(binaryFile);var instance=await WebAssembly.instantiate(binary,imports);return instance}catch(reason){err(`failed to asynchronously prepare wasm: ${reason}`);abort(reason)}}async function instantiateAsync(binary,binaryFile,imports){if(!binary&&!isFileURI(binaryFile)&&!ENVIRONMENT_IS_N[..]
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                

RuntimeError: Aborted(segmentation fault). Build with -sASSERTIONS for more info.
    at abort (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:4124)
    at segfault (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:3011)
    at wasm://wasm/00054d52:wasm-function[215]:0x12bad
    at wasm://wasm/00054d52:wasm-function[13]:0x623
    at initRuntime (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:3729)
    at doRun (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:13856)
    at run (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:14133)
    at removeRunDependency (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:6830)
    at receiveInstance (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:5595)
    at receiveInstantiationResult (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:5742)
    at createWasm (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:6067)
Thrown at:
    at abort (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:4124)
    at segfault (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:3011)
    at $func215 (wasm://wasm/00054d52:1:76718)
    at $__wasm_call_ctors (wasm://wasm/00054d52:1:1572)
    at initRuntime (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:3729)
    at doRun (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:13856)
    at run (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:14133)
    at removeRunDependency (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:6830)
    at receiveInstance (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:5595)
    at receiveInstantiationResult (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:5742)


Node.js v22.16.0
-- end program output --


======================================================================
FAIL: test_wrap_malloc_mimalloc (test_core.core2s.test_wrap_malloc_mimalloc)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\emsdk\emscripten\main\test\common.py", line 377, in resulting_test
    return func(self, *args)
  File "C:\emsdk\emscripten\main\test\test_core.py", line 337, in decorated
    return func(self, *args, **kwargs)
  File "C:\emsdk\emscripten\main\test\test_core.py", line 352, in decorated
    return func(self, *args, **kwargs)
  File "C:\emsdk\emscripten\main\test\test_core.py", line 8661, in test_wrap_malloc
    self.do_runf('core/test_wrap_malloc.c', 'OK.', cflags=args)
    ~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\emsdk\emscripten\main\test\common.py", line 1387, in do_runf
    return self._build_and_run(filename, expected_output, **kwargs)
           ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\emsdk\emscripten\main\test\common.py", line 1435, in _build_and_run
    js_output = self.run_js(js_file, engine, args,
                            input=input,
                            assert_returncode=assert_returncode,
                            interleaved_output=interleaved_output)
  File "C:\emsdk\emscripten\main\test\common.py", line 1019, in run_js
    self.fail('JS subprocess failed (%s): %s (expected=%s).  Output:\n%s' % (error.cmd, error.returncode, assert_returncode, ret))
    ~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: JS subprocess failed (C:/emsdk/node/22.16.0_64bit/bin/node.exe --stack-trace-limit=50 --trace-uncaught C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js): 1 (expected=0).  Output:
Aborted(segmentation fault)
C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1
var Module=typeof Module!="undefined"?Module:{};var ENVIRONMENT_IS_WEB=!!globalThis.window;var ENVIRONMENT_IS_WORKER=!!globalThis.WorkerGlobalScope;var ENVIRONMENT_IS_NODE=globalThis.process?.versions?.node&&globalThis.process?.type!="renderer";var arguments_=[];var thisProgram="./this.program";var quit_=(status,toThrow)=>{throw toThrow};var _scriptName=globalThis.document?.currentScript?.src;if(typeof __filename!="undefined"){_scriptName=__filename}else if(ENVIRONMENT_IS_WORKER){_scriptName=self.location.href}var scriptDirectory="";function locateFile(path){if(Module["locateFile"]){return Module["locateFile"](path,scriptDirectory)}return scriptDirectory+path}var readAsync,readBinary;if(ENVIRONMENT_IS_NODE){var fs=require("node:fs");scriptDirectory=__dirname+"/";readBinary=filename=>{filename=isFileURI(filename)?new URL(filename):filename;var ret=fs.readFileSync(filename);return ret};readAsync=async(filename,binary=true)=>{filename=isFileURI(filename)?new URL(filename):filename;var ret=fs.readFileSync(filename,binary?undefined:"utf8");return ret};if(process.argv.length>1){thisProgram=process.argv[1].replace(/\\/g,"/")}arguments_=process.argv.slice(2);if(typeof module!="undefined"){module["exports"]=Module}quit_=(status,toThrow)=>{process.exitCode=status;throw toThrow}}else if(ENVIRONMENT_IS_WEB||ENVIRONMENT_IS_WORKER){try{scriptDirectory=new URL(".",_scriptName).href}catch{}{if(ENVIRONMENT_IS_WORKER){readBinary=url=>{var xhr=new XMLHttpRequest;xhr.open("GET",url,false);xhr.responseType="arraybuffer";xhr.send(null);return new Uint8Array(xhr.response)}}readAsync=async url=>{if(isFileURI(url)){return new Promise((resolve,reject)=>{var xhr=new XMLHttpRequest;xhr.open("GET",url,true);xhr.responseType="arraybuffer";xhr.onload=()=>{if(xhr.status==200||xhr.status==0&&xhr.response){resolve(xhr.response);return}reject(xhr.status)};xhr.onerror=reject;xhr.send(null)})}var response=await fetch(url,{credentials:"same-origin"});if(response.ok){return response.arrayBuffer()}throw new Error(response.status+" : "+response.url)}}}else{}var out=console.log.bind(console);var err=console.error.bind(console);var wasmBinary;var ABORT=false;var EXITSTATUS;var isFileURI=filename=>filename.startsWith("file://");class EmscriptenEH{}class EmscriptenSjLj extends EmscriptenEH{}function SAFE_HEAP_INDEX(arr,idx,action){const bytes=arr.BYTES_PER_ELEMENT;const dest=idx*bytes;if(idx<=0)abort(`segmentation fault ${action} ${bytes} bytes at address ${dest}`);if(runtimeInitialized){var brk=_sbrk(0);if(dest+bytes>brk)abort(`segmentation fault, exceeded the top of the available dynamic heap when ${action} ${bytes} bytes at address ${dest}. DYNAMICTOP=${brk}`);if(brk<_emscripten_stack_get_base())abort(`brk >= _emscripten_stack_get_base() (brk=${brk}, _emscripten_stack_get_base()=${_emscripten_stack_get_base()})`);if(brk>wasmMemory.buffer.byteLength)abort(`brk <= wasmMemory.buffer.byteLength (brk=${brk}, wasmMemory.buffer.byteLength=${wasmMemory.buffer.byteLength})`)}return idx}function segfault(){abort("segmentation fault")}function alignfault(){abort("alignment fault")}var runtimeInitialized=false;function updateMemoryViews(){var b=wasmMemory.buffer;HEAP8=new Int8Array(b);HEAP16=new Int16Array(b);HEAPU8=new Uint8Array(b);HEAPU16=new Uint16Array(b);HEAP32=new Int32Array(b);HEAPU32=new Uint32Array(b);HEAPF32=new Float32Array(b);HEAPF64=new Float64Array(b);HEAP64=new BigInt64Array(b);HEAPU64=new BigUint64Array(b)}function preRun(){if(Module["preRun"]){if(typeof Module["preRun"]=="function")Module["preRun"]=[Module["preRun"]];while(Module["preRun"].length){addOnPreRun(Module["preRun"].shift())}}callRuntimeCallbacks(onPreRuns)}function initRuntime(){runtimeInitialized=true;wasmExports["__wasm_call_ctors"]()}function preMain(){}function postRun(){if(Module["postRun"]){if(typeof Module["postRun"]=="function")Module["postRun"]=[Module["postRun"]];while(Module["postRun"].length){addOnPostRun(Module["postRun"].shift())}}callRuntimeCallbacks(onPostRuns)}function abort(what){Module["onAbort"]?.(what);what=`Aborted(${what})`;err(what);ABORT=true;what+=". Build with -sASSERTIONS for more info.";var e=new WebAssembly.RuntimeError(what);throw e}var wasmBinaryFile;function findWasmBinary(){return locateFile("test_wrap_malloc.wasm")}function getBinarySync(file){if(file==wasmBinaryFile&&wasmBinary){return new Uint8Array(wasmBinary)}if(readBinary){return readBinary(file)}throw"both async and sync fetching of the wasm failed"}async function getWasmBinary(binaryFile){if(!wasmBinary){try{var response=await readAsync(binaryFile);return new Uint8Array(response)}catch{}}return getBinarySync(binaryFile)}async function instantiateArrayBuffer(binaryFile,imports){try{var binary=await getWasmBinary(binaryFile);var instance=await WebAssembly.instantiate(binary,imports);return instance}catch(reason){err(`failed to asynchronously prepare wasm: ${reason}`);abort(reason)}}async function instantiateAsync(binary,binaryFile,imports){if(!binary&&!isFileURI(binaryFile)&&!ENVIRONMENT_IS_N[..]
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                

RuntimeError: Aborted(segmentation fault). Build with -sASSERTIONS for more info.
    at abort (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:4124)
    at segfault (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:3011)
    at wasm://wasm/00054d52:wasm-function[215]:0x12bad
    at wasm://wasm/00054d52:wasm-function[13]:0x623
    at initRuntime (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:3729)
    at doRun (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:13856)
    at run (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:14133)
    at removeRunDependency (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:6830)
    at receiveInstance (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:5595)
    at receiveInstantiationResult (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:5742)
    at createWasm (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:6067)
Thrown at:
    at abort (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:4124)
    at segfault (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:3011)
    at $func215 (wasm://wasm/00054d52:1:76718)
    at $__wasm_call_ctors (wasm://wasm/00054d52:1:1572)
    at initRuntime (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:3729)
    at doRun (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:13856)
    at run (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:14133)
    at removeRunDependency (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:6830)
    at receiveInstance (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:5595)
    at receiveInstantiationResult (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:5742)


Node.js v22.16.0


Stdout:
-- begin program output --
Aborted(segmentation fault)
C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1
var Module=typeof Module!="undefined"?Module:{};var ENVIRONMENT_IS_WEB=!!globalThis.window;var ENVIRONMENT_IS_WORKER=!!globalThis.WorkerGlobalScope;var ENVIRONMENT_IS_NODE=globalThis.process?.versions?.node&&globalThis.process?.type!="renderer";var arguments_=[];var thisProgram="./this.program";var quit_=(status,toThrow)=>{throw toThrow};var _scriptName=globalThis.document?.currentScript?.src;if(typeof __filename!="undefined"){_scriptName=__filename}else if(ENVIRONMENT_IS_WORKER){_scriptName=self.location.href}var scriptDirectory="";function locateFile(path){if(Module["locateFile"]){return Module["locateFile"](path,scriptDirectory)}return scriptDirectory+path}var readAsync,readBinary;if(ENVIRONMENT_IS_NODE){var fs=require("node:fs");scriptDirectory=__dirname+"/";readBinary=filename=>{filename=isFileURI(filename)?new URL(filename):filename;var ret=fs.readFileSync(filename);return ret};readAsync=async(filename,binary=true)=>{filename=isFileURI(filename)?new URL(filename):filename;var ret=fs.readFileSync(filename,binary?undefined:"utf8");return ret};if(process.argv.length>1){thisProgram=process.argv[1].replace(/\\/g,"/")}arguments_=process.argv.slice(2);if(typeof module!="undefined"){module["exports"]=Module}quit_=(status,toThrow)=>{process.exitCode=status;throw toThrow}}else if(ENVIRONMENT_IS_WEB||ENVIRONMENT_IS_WORKER){try{scriptDirectory=new URL(".",_scriptName).href}catch{}{if(ENVIRONMENT_IS_WORKER){readBinary=url=>{var xhr=new XMLHttpRequest;xhr.open("GET",url,false);xhr.responseType="arraybuffer";xhr.send(null);return new Uint8Array(xhr.response)}}readAsync=async url=>{if(isFileURI(url)){return new Promise((resolve,reject)=>{var xhr=new XMLHttpRequest;xhr.open("GET",url,true);xhr.responseType="arraybuffer";xhr.onload=()=>{if(xhr.status==200||xhr.status==0&&xhr.response){resolve(xhr.response);return}reject(xhr.status)};xhr.onerror=reject;xhr.send(null)})}var response=await fetch(url,{credentials:"same-origin"});if(response.ok){return response.arrayBuffer()}throw new Error(response.status+" : "+response.url)}}}else{}var out=console.log.bind(console);var err=console.error.bind(console);var wasmBinary;var ABORT=false;var EXITSTATUS;var isFileURI=filename=>filename.startsWith("file://");class EmscriptenEH{}class EmscriptenSjLj extends EmscriptenEH{}function SAFE_HEAP_INDEX(arr,idx,action){const bytes=arr.BYTES_PER_ELEMENT;const dest=idx*bytes;if(idx<=0)abort(`segmentation fault ${action} ${bytes} bytes at address ${dest}`);if(runtimeInitialized){var brk=_sbrk(0);if(dest+bytes>brk)abort(`segmentation fault, exceeded the top of the available dynamic heap when ${action} ${bytes} bytes at address ${dest}. DYNAMICTOP=${brk}`);if(brk<_emscripten_stack_get_base())abort(`brk >= _emscripten_stack_get_base() (brk=${brk}, _emscripten_stack_get_base()=${_emscripten_stack_get_base()})`);if(brk>wasmMemory.buffer.byteLength)abort(`brk <= wasmMemory.buffer.byteLength (brk=${brk}, wasmMemory.buffer.byteLength=${wasmMemory.buffer.byteLength})`)}return idx}function segfault(){abort("segmentation fault")}function alignfault(){abort("alignment fault")}var runtimeInitialized=false;function updateMemoryViews(){var b=wasmMemory.buffer;HEAP8=new Int8Array(b);HEAP16=new Int16Array(b);HEAPU8=new Uint8Array(b);HEAPU16=new Uint16Array(b);HEAP32=new Int32Array(b);HEAPU32=new Uint32Array(b);HEAPF32=new Float32Array(b);HEAPF64=new Float64Array(b);HEAP64=new BigInt64Array(b);HEAPU64=new BigUint64Array(b)}function preRun(){if(Module["preRun"]){if(typeof Module["preRun"]=="function")Module["preRun"]=[Module["preRun"]];while(Module["preRun"].length){addOnPreRun(Module["preRun"].shift())}}callRuntimeCallbacks(onPreRuns)}function initRuntime(){runtimeInitialized=true;wasmExports["__wasm_call_ctors"]()}function preMain(){}function postRun(){if(Module["postRun"]){if(typeof Module["postRun"]=="function")Module["postRun"]=[Module["postRun"]];while(Module["postRun"].length){addOnPostRun(Module["postRun"].shift())}}callRuntimeCallbacks(onPostRuns)}function abort(what){Module["onAbort"]?.(what);what=`Aborted(${what})`;err(what);ABORT=true;what+=". Build with -sASSERTIONS for more info.";var e=new WebAssembly.RuntimeError(what);throw e}var wasmBinaryFile;function findWasmBinary(){return locateFile("test_wrap_malloc.wasm")}function getBinarySync(file){if(file==wasmBinaryFile&&wasmBinary){return new Uint8Array(wasmBinary)}if(readBinary){return readBinary(file)}throw"both async and sync fetching of the wasm failed"}async function getWasmBinary(binaryFile){if(!wasmBinary){try{var response=await readAsync(binaryFile);return new Uint8Array(response)}catch{}}return getBinarySync(binaryFile)}async function instantiateArrayBuffer(binaryFile,imports){try{var binary=await getWasmBinary(binaryFile);var instance=await WebAssembly.instantiate(binary,imports);return instance}catch(reason){err(`failed to asynchronously prepare wasm: ${reason}`);abort(reason)}}async function instantiateAsync(binary,binaryFile,imports){if(!binary&&!isFileURI(binaryFile)&&!ENVIRONMENT_IS_N[..]
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                

RuntimeError: Aborted(segmentation fault). Build with -sASSERTIONS for more info.
    at abort (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:4124)
    at segfault (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:3011)
    at wasm://wasm/00054d52:wasm-function[215]:0x12bad
    at wasm://wasm/00054d52:wasm-function[13]:0x623
    at initRuntime (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:3729)
    at doRun (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:13856)
    at run (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:14133)
    at removeRunDependency (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:6830)
    at receiveInstance (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:5595)
    at receiveInstantiationResult (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:5742)
    at createWasm (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:6067)
Thrown at:
    at abort (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:4124)
    at segfault (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:3011)
    at $func215 (wasm://wasm/00054d52:1:76718)
    at $__wasm_call_ctors (wasm://wasm/00054d52:1:1572)
    at initRuntime (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:3729)
    at doRun (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:13856)
    at run (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:14133)
    at removeRunDependency (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:6830)
    at receiveInstance (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:5595)
    at receiveInstantiationResult (C:\emsdk\emscripten\main\out\test\test_wrap_malloc.js:1:5742)


Node.js v22.16.0
-- end program output --

----------------------------------------------------------------------
Ran 1 test in 1.634s

FAILED (failures=1)
```

The failure points to

```
10f62b4c1a529324fd2bd4b1f6359b3708f37b91 is the first bad commit
commit 10f62b4c1a529324fd2bd4b1f6359b3708f37b91
Author: Kleis Auke Wolthuizen <github@kleisauke.nl>
Date:   Tue Apr 21 19:56:21 2026 +0200

    Update mimalloc to 3.3.1 (#26696)

    Fixes: #23090.
    Supersedes: #23037.
    Supersedes: #24542.

 ChangeLog.md                                       |    1 +
 system/lib/mimalloc/LICENSE                        |    2 +-
 system/lib/mimalloc/README.emscripten              |    2 +-
 system/lib/mimalloc/include/mimalloc-override.h    |   20 +-
 system/lib/mimalloc/include/mimalloc-stats.h       |  164 ++
 system/lib/mimalloc/include/mimalloc.h             |  336 ++-
 system/lib/mimalloc/include/mimalloc/atomic.h      |  300 +-
 system/lib/mimalloc/include/mimalloc/bits.h        |  340 +++
 system/lib/mimalloc/include/mimalloc/internal.h    | 1347 +++++----
 system/lib/mimalloc/include/mimalloc/prim.h        |  407 ++-
 system/lib/mimalloc/include/mimalloc/track.h       |   25 +-
 system/lib/mimalloc/include/mimalloc/types.h       |  983 +++----
 system/lib/mimalloc/readme.md                      |  474 ++--
 system/lib/mimalloc/src/alloc-aligned.c            |  352 ++-
 system/lib/mimalloc/src/alloc-override.c           |  138 +-
 system/lib/mimalloc/src/alloc-posix.c              |   65 +-
 system/lib/mimalloc/src/alloc.c                    |  602 ++--
 system/lib/mimalloc/src/arena-meta.c               |  179 ++
 system/lib/mimalloc/src/arena.c                    | 2921 +++++++++++++++-----
 system/lib/mimalloc/src/bitmap.c                   | 2233 ++++++++++++---
 system/lib/mimalloc/src/bitmap.h                   |  378 ++-
 system/lib/mimalloc/src/free.c                     |  590 ++--
 system/lib/mimalloc/src/heap.c                     |  786 ++----
 system/lib/mimalloc/src/init.c                     | 1374 ++++++---
 system/lib/mimalloc/src/libc.c                     |  264 +-
 system/lib/mimalloc/src/options.c                  |  371 ++-
 system/lib/mimalloc/src/os.c                       |  554 ++--
 system/lib/mimalloc/src/page-map.c                 |  447 +++
 system/lib/mimalloc/src/page-queue.c               |  301 +-
 system/lib/mimalloc/src/page.c                     |  831 +++---
 system/lib/mimalloc/src/prim/emscripten/prim.c     |   39 +-
 .../mimalloc/src/prim/osx/alloc-override-zone.c    |   18 +-
 system/lib/mimalloc/src/prim/prim.c                |   49 +
 system/lib/mimalloc/src/prim/unix/prim.c           |  281 +-
 system/lib/mimalloc/src/prim/wasi/prim.c           |   49 +-
 system/lib/mimalloc/src/prim/windows/prim.c        |  627 ++++-
 system/lib/mimalloc/src/random.c                   |   46 +-
 system/lib/mimalloc/src/segment-map.c              |  155 --
 system/lib/mimalloc/src/segment.c                  | 1524 ----------
 system/lib/mimalloc/src/static.c                   |   10 +-
 system/lib/mimalloc/src/stats.c                    |  724 +++--
 system/lib/mimalloc/src/theap.c                    |  714 +++++
 system/lib/mimalloc/src/threadlocal.c              |  227 ++
 test/test_core.py                                  |    4 +-
 test/test_other.py                                 |    4 +-
 tools/system_libs.py                               |   16 +-
 46 files changed, 14114 insertions(+), 7160 deletions(-)
 create mode 100644 system/lib/mimalloc/include/mimalloc-stats.h
 create mode 100644 system/lib/mimalloc/include/mimalloc/bits.h
 create mode 100644 system/lib/mimalloc/src/arena-meta.c
 create mode 100644 system/lib/mimalloc/src/page-map.c
 delete mode 100644 system/lib/mimalloc/src/segment-map.c
 delete mode 100644 system/lib/mimalloc/src/segment.c
 create mode 100644 system/lib/mimalloc/src/theap.c
 create mode 100644 system/lib/mimalloc/src/threadlocal.c
```

i.e. #26696 . CC @kleisauke

--- a/tools/system_libs.py
+++ b/tools/system_libs.py
@@ -1889,8 +1889,6 @@ class libmimalloc(MTLibrary):
     '-DNDEBUG',
     # Emscripten uses musl libc internally
     '-DMI_LIBC_MUSL',
-    # enable use of `__builtin_thread_pointer()`
-    '-DMI_USE_BUILTIN_THREAD_POINTER',
   ]
 
   # malloc/free/calloc are runtime functions and can be generated during LTO
@@ -1912,6 +1910,13 @@ class libmimalloc(MTLibrary):
   src_files += [utils.path_from_root('system/lib/emmalloc.c')]
   # Include sbrk.c in libc, it uses tracing and libc itself doesn't have a tracing variant.
   src_files += [utils.path_from_root('system/lib/libc/sbrk.c')]
+
+  def get_cflags(self):
+    cflags = super().get_cflags()
+    if self.is_mt:
+      # enable use of `__builtin_thread_pointer()` in multithreaded builds
+      cflags += ['-DMI_USE_BUILTIN_THREAD_POINTER']
+    return cflags
 
   def can_use(self):
     return super().can_use() and settings.MALLOC == 'mimalloc'

